pkg/controller/storageprofile-controller.go

```diff
@@ -141,7 +141,13 @@ func (r *StorageProfileReconciler) reconcileAccessModes(sc *storagev1.StorageCla
 }
 
 func (r *StorageProfileReconciler) createEmptyStorageProfile(sc *storagev1.StorageClass) (*cdiv1.StorageProfile, error) {
-	storageProfile := MakeEmptyStorageProfileSpec(sc.Name)
+	var storageProfile *cdiv1.StorageProfile
+	if sc.Annotations["cdi.kubevirt.io/storage-profile"] == "host-assisted" {
```

Comment:
The annotation should be in a constant.  Also, since there is more to a storage profile than just the clone strategy I think we should use "cdi.kubevirt.io/clonestrategy".  Since the term used in CDI is "copy" we should prefer that to "host-assisted".

Reply:
This code seems to only handle the case when the storage profile does not exist.  It does not handle adding the annotation to the SC when the profile exists.  It also does not handle when the annotation is removed (should set clone strategy back to default)

Reply:
@mhenriks this change, sets the cloneStrategy as host assisted if a storageClass has an annotation of cdi.kubevirt.io/storage-profile=host-assisted
This way any vendors(even non ODF) who wants to do a host-assisted clone, can annotate their storageClass with the above annotation and CDI would do a host-assisted one, so we do not need to worry about adding the annotation  to the StorageClass, based on the StorageProfile, since storageProfile would only execute Host-assisted, if StorageClass has that annotaion.

And since this logic resides under the reconcile loop, if the annotation is  removed, then it would execute the alternate condition and proceed with Line 149[https://github.com/kubevirt/containerized-data-importer/pull/2004/commits/23c033115a4c88e879bee03169c7bbd27327d60e#diff-3a3e2d2d37a456bef2fad553faed8c75f9fecd62ad31cc3fbe8179ba670fd3eaR149]

This is my understanding. Please correct me if this is wrong. :-)  @aglitke @mhenriks

Reply:
it looks like you have added the code in the place which is only invoked at profile init time,
```
r.getStorageProfile(sc) -> 
  	**if k8serrors.IsNotFound(err) {** 
	  			storageProfile, err = r.createEmptyStorageProfile(sc)
	 }
}
...
```
The correct place would be to add new code in a main recocile loop in line 61. You can add `r.reconcileCloneStrategy()` the same way we have `reconcileAccessModes` and `reconcileVolumeMode`. 

Anyway, it would be better to rebase first, as a change to storageProfile reconciliation was merged last week. That PR also added a series of unit tests `storageprofile-controller_test.go` - which might be really helpful in experimenting, and now it will be really easy to add new test.

Reply:
@pkesavap and I totally agree that implementing it with an annotation is a good - non intrusive and vendor agnostic way. I just think that `cdi.kubevirt.io/clonestrategy` is better than `cdi.kubevirt.io/storage-profile`.